2.x/SCO/x-oca-asset.md

```diff
@@ -0,0 +1,289 @@
+| **ingress** | `x-oca-traffic-type` | Specifies information like interface number and name, vlan, and zone information to classify ingress traffic using the Traff9c-type sub-object.|
+| **egress** | `x-oca-traffic-type` | Specifies information like interface number and name, vlan, and zone information to classify egress traffic using the Traffic-type sub object.|
+| **geo_ref**|`object-ref`| references the geolocation of this host. must be of type `x-oca-geo`|
+
+### Container Extenstion
+
+Extension Name `x-oca-container-ext`
+
+The container asset extension represents a container.
+
+| property name | type | description |
+|--|--|--|
+| **type** | `string` | x-oca-container-ext |
+| **name** | `string` | container name |
+| **container_id** | `string` | container id |
+| **image_name** | `string` | container image name |
+| **image_id** | `string` | container image id |
+| **container_type** | `container_type_ov` | container type | 
+| **privileged** | `boolean` | indicates a priviliged container |
+
+#### Container Type Vocabluary
+
+Vocabulary Name: `container-type-ov`
+
+An open vocabulary for the container_type field of the container extension of an asset object.
+
+| Vocabulary Value | Description |
+|--|--|
+| `docker` | Specifies a docker container |
+| `crio` | Specifies a crio container |
+| `mesos` | Specifies a mesos container |
+| `lxc` | Specifies a LXC container |
+| `libvirt_lxc` | Specifies a libvirt LXC container |
+| `rkt` | Specifies a rkt container |
+| `oci` | Specifies a oci standards based container |
+| `custom` |Specifies a custom container |
+
+### Pod Extenstion
+
+Extension Name `x-oca-pod-ext`
+
+The pod asset extension represents an oc/k8s pod.
+
+| property name | type | description |
+|--|--|--|
+| **type** | `string` | x-oca-pod-ext |
+| **name** | `string` | pod name |
+| **ip_refs** | `list` of type `object-ref` | references the ip addresses allocated to this pod. must be `ipv4-addr` or `ipv6-addr` |
+
+#### Traffic Type Sub-Object
+
+Type Name `x-oca-traffic-type`
+
+| value | type | description |
+|--|--|--|
+| **zone** | `string` | Specifies the network zone of the inbound or outbound traffic.|
+| **interfaces** | list of type `x-oca-interface-type` | Specifies the interface information. Each dictionary key SHOULD come from the Interface vocabulary. |
+
+#### Interface Type Sub-Object
+
+Type Name `x-oca-interface-type`
+
+| value | type | description |
+|--|--|--|
+| **alias** | `string` | Specifies the interface alias as reported by the system. |
+| **interface_id** | `string` | Specifies the interface ID. |
+| **name** | `string` | Specifies the interface name as reported by the system.|
+
+### Example 1
+
+```json
+    {
+	    "0":
+	    {
+	        "type": "ipv4-addr",
+	        "value": "9.9.9.9"
+	    },
+	    "1":
+	    {
+	        "type": "x-oca-asset",
+	        "ip_refs": ["0", "2"],
+	        "hostname": "example host",
+			    "host_id": "123A",
```

Comment:
Is this just git being weird or do these have odd tab spacing?

Reply:
Aah yes, in your examples, which i copied originally, and then edited, there was a lot of hand-created different types of spacing. We should never use spacing in a json, as it will visualise differently on different formats (IDE vs web).

I had to feed all of my stuff through online space eliminators for json files. Then you use the auto json layout of your IDE, then update it that way. But unfortunately i did not fix any spacing in the original v2.0 examples, so they are still there.

I manually removed these spaces you indicate, which should fix the problem's you have highlighted